pkg/webui/console/views/application-overview/application-overview.js

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Col, Row, Container } from 'react-grid-system'
+
+import sharedMessages from '../../../lib/shared-messages'
+import PropTypes from '../../../lib/prop-types'
+import PAGE_SIZES from '../../constants/page-sizes'
+
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import DateTime from '../../../lib/components/date-time'
+import DevicesTable from '../../containers/devices-table'
+import DataSheet from '../../../components/data-sheet'
+import ApplicationEvents from '../../containers/application-events'
+import EntityTitleSection from '../../components/entity-title-section'
+import KeyValueTag from '../../components/key-value-tag'
+import Status from '../../../components/status'
+import Spinner from '../../../components/spinner'
+import Message from '../../../lib/components/message'
+import withFeatureRequirement from '../../lib/components/with-feature-requirement'
+import withRequest from '../../../lib/components/with-request'
+
+import { mayViewApplicationInfo } from '../../lib/feature-checks'
+
+import style from './application-overview.styl'
+
+@withRequest(({ appId, loadData }) => loadData(appId), () => false)
+@withFeatureRequirement(mayViewApplicationInfo, {
+  redirect: '/',
+})
+class ApplicationOverview extends React.Component {
+  static propTypes = {
+    apiKeysTotalCount: PropTypes.number,
+    appId: PropTypes.string.isRequired,
+    application: PropTypes.application.isRequired,
+    collaboratorsTotalCount: PropTypes.number,
+    devicesTotalCount: PropTypes.number,
+    link: PropTypes.bool,
+    statusBarFetching: PropTypes.bool.isRequired,
+  }
+
+  static defaultProps = {
+    collaboratorsTotalCount: undefined,
+    apiKeysTotalCount: undefined,
+    devicesTotalCount: undefined,
```

Comment:
If these are set to `undefined` the `value` gets initially passed as `undefined` and that makes `KeyValue` complain about required prop. setting to 0 would work.